.github/workflows/main.yml

```
@@ -320,11 +320,11 @@ jobs:
       with:
         command: |
           ctDir = getenv('CANTERA_ROOT');
-          ctToolboxDir = fullfile(ctDir, 'interfaces', 'matlab_experimental');
+          ctToolboxDir = fullfile(ctDir, 'interfaces', 'matlab');
           ctIncludeDir = fullfile(ctDir, 'interfaces', 'clib', 'include');
           ctLibDir = fullfile(ctDir, 'build', 'lib');
           addpath(genpath(ctToolboxDir));
-          ctBuildInterface(ctToolboxDir, ctIncludeDir, ctLibDir);
+          ct.buildInterface(ctToolboxDir, ctIncludeDir, ctLibDir);
```

```suggestion
          ct.impl.buildInterface(ctToolboxDir, ctIncludeDir, ctLibDir);
```
I think this should go into the 'implementation' folder to avoid confusion. As an aside, I'd also prefer `clib` to `impl` for the sub-namespace.

---

I prefer `impl` because the fact that it's a C-library is somewhat irrelevant... we could (hypothetically) replace it with C++ or Python and leave the `impl` namespace to avoid breaking user code.

---

> I prefer `impl` because the fact that it's a C-library is somewhat irrelevant... we could (hypothetically) replace it with C++ or Python and leave the `impl` namespace to avoid breaking user code.

Interesting thought. While I don't anticipate a move away from CLib, the folder will likely remain untouched by users (it's pretty much what's 'under the hood'), so breaking user code is largely hypothetical. I still dislike `impl`, though. What about `core`?

---

The purpose of the nested `impl` namespace is as a place to hide the implementation details -- the user should never call any of these functions directly. I read some suggestions that having a `private` folder within a namespace provided a way of doing this, but I couldn't get that to work so ended up with this instead.

`buildInterface`, however, is not a private implementation detail, as it needs to be called at some point (even if only once ever).

---

I’m totally on-board with ‘hiding’ these details, but don’t think that this needs to be considered private. I’m fine with users calling in directly (caveat emptor!). Regarding the right space for `buildInterface` my main thought was to hide that in the same place where the rest of the ‘under the hood’ functions live.

---

`impl` or `core` seems fine to me. As @speth said, let's hide the implementation details, including the name.